src/3.1/docker-entrypoint.sh

```diff
@@ -2,10 +2,76 @@
 
 cmd="$1"
 
+# We intentionally chown /data to root in Dockerfile so we can detect
+# if no volume is mounted
+if [[ "$(stat -c %u /data)" != "0" ]]; then
+  # /data is a volume, is the same uid/gid for neo4j user
+  user_uid="$(stat -c %u /data)"
+  user_gid="$(stat -c %g /data)"
+elif [ -d /conf ] && [[ "${cmd}" == "dump-config" ]]; then
+  # A configuration volume has been mounted and we are dumping config
+  user_uid="$(stat -c %u /conf)"
+  user_gid="$(stat -c %g /conf)"
+else
+  # Set to zero (root) to signal no mounting has been done
```

Comment:
Using `0` as a sentinel value here is misleading, since we never actually want to use `root`. Can you just leave them unset and test against that instead?

Reply:
Actually, it serves a dual-purpose. In the special case that a user would have mounted a volume which was actually a directory owned by root, then this approach will make sure we do in fact run as a non-root user. So using `0` as the sentinel value makes sure we never run under that uid.

We can of course add an additional check to the ifs later to check if it's `unset or zero`. If you think that's cleaner I'll change it.

Writing the reply has convinced me that it's cleaner so I'll change it...